### ISSUE TITLE
Stop connecting from KSQL DB Server to Confluent Cloud

### ISSUE THREAD

**Issue by vidanimegh** [confluentinc/ksql#9692]
**Search for existing answers**
Many questions have been asked before.  Please have a quick search to see if there is already an answer to your question.  There are two main resources for finding answers:

1. [Existing KSQL Github issues, including _closed_ issues](https://github.com/confluentinc/ksql/issues?utf8=%E2%9C%93&q=is%3Aissue+)
1. [KSQL on Stack Overflow](https://stackoverflow.com/questions/tagged/ksql)

If you can't see an answer to your question, then please raise a new Github or Stack overflow issue.

**Provide details of the setup you're running**
KSQLDB version - 0.23.1
OS - CentOS 7.9 

**Outline your question**

I've searched a bit around this and couldn't find much information so far.

Basically, I want to understand whether there is a default behaviour in ksqldb by which the servers keep trying to connect with the confluent cloud?

I have done an in-house ksqldb cluster setup using docker and my network security team is saying that they have found that there are continuous outgoing requests from my servers outside the company domain.

Under what circumstances does ksqldb try to connect with confluent cloud? Is there a way to disable that behaviour?

Thanks,
Megh

**Comment by suhas-satish** [comment]
@vidanimegh , can you post some examples from logs or stacktrace about which are those outbound connections from ksql server to confluent cloud ?

**Comment by vidanimegh** [comment]
Hi @suhas-satish , I'm not seeing any log entries that suggests that. But I have a tcpdump from the ksqldb container that tries to do DNS lookup to Google's DNS servers 8.8.8.8

> 06:53:39.618065 IP <myhostname1>.<port1> > 8.8.8.8.domain: 4835+ A? ksqldb-server-3. (33)
E..=..@.?...
..<.......5.)...............ksqldb-server-3.....
06:53:39.913211 IP <myhostname2>.<port2> > 8.8.8.8.domain: 2236+ A? ksqldb-server-2. (33)
E..=..@.?..Q
..;.....g.5.)...............ksqldb-server-2.....
06:53:40.117884 IP <myhostname3>.<por3> > 8.8.8.8.domain: 16680+ A? ksqldb-server-2. (33)
E..=.1@.?..*
..<.......5.)U#A(...........ksqldb-server-2.....
06:53:40.975570 IP <myhostname3>.<port4> > 8.8.8.8.domain: 51109+ A? ksqldb-server-3. (33)
E..=..@.?...
..;.......5.)...............ksqldb-server-3.....
06:53:41.175938 IP <myhostname4>.<port5> > 8.8.8.8.domain: 65514+ A? ksqldb-server-2. (33)
E..=.T@.?...
..;.......5.)...............ksqldb-server-2.....
06:53:41.213535 IP <myhostname5>.<port6> > 8.8.8.8.domain: 47129+ A? ksqldb-server-5. (33)
E..=..@.?..K
..;.......5.).".............ksqldb-server-5.....

These are ksqldb containers so I'm suspecting that they're trying to reach some hostname outside our domain (possibly Confluent cloud) and not able to resolve the hostname so trying with google DNS.

Thanks,
Megh

**Comment by vidanimegh** [comment]
Went inside the docker and checked the contents of /etc/resolv.conf
>search local <mydomain>
nameserver 127.0.0.11
options ndots:0

This is what I see. Nothing to suggest why it should look up Google DNS.

**Comment by vidanimegh** [comment]
@suhas-satish any observations?

**Comment by suhas-satish** [comment]
@vidanimegh , there is a healthcheck service that might also go over the internet. Is it possible thats making these callouts?

**Comment by vidanimegh** [comment]
@suhas-satish You were right. This was happening because of the heartbeats!

Since I have a docker setup in 3 different servers, and they are not in the same docker network by default, the name resolution was going to Google DNS.

To fix it, I added extra_hosts property in the docker-compose.yaml and gave "hostname:ip" entries for all the ksqldb-server instances.

Also, after this I discovered that hostname resolution was okay but there were still calls going to Google DNS for resolution of  version-check.io.confluent.local

To fix that, I added
KSQL_OPTS: "-Dconfluent.support.metrics.enable=false"

This has solved my problem!

Thanks for the help.

Megh
